src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -6053,5 +6053,36 @@ public async Task M(object o)
             var comp = CSharpTestBase.CreateCompilation(source);
             comp.VerifyEmitDiagnostics();
         }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void Repro()
```

Comment:
>Repro [](http://example.com/codeflow?start=20&length=5)

We probably need a more meaningful name for the test. Also, could you provide details about the nature of the failure for this scenario?

Reply:
In the presence of pattern-matching, the capture walker did not consider the enumerator temp as a capture and it remained a local. Also related: https://github.com/dotnet/roslyn/pull/72273#discussion_r1514005305

Reply:
>> Also, could you provide details about the nature of the failure for this scenario?

> In the presence of pattern-matching, the capture walker did not consider the enumerator temp as a capture and it remained a local. 

And how does this affect the test scenario? Does the test crash (where?), asserts (where?), exhibits unexpected behavior at runtime (what behavior), something else?

Reply:
Since the enumerator is a local it's left unassigned when the continuation runs at runtime, leading to an NRE.

![image](https://github.com/dotnet/roslyn/assets/3105979/d3e23c60-ab71-4957-ba5b-ac085a99abe3)